src/main/RemoteRequest.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * RemoteRequest is used to for http requests on a remote server which can be
+ * fetched in chunks, e.g. using a Range request.
+ * @flow
+ */
+'use strict';
+
+import Q from 'q';
+
+type Chunk = {
+  start: number;
+  stop: number;
+  buffer: Object; // TODO: more efficient data store
```

Comment:
I think the folder structure you have in mind makes sense @akmorrow13  if you will be using this class as a utility class rather than extending it. If the latter, it might make sense to make it more abstract and move it under the `source` folder.